packages/lodestar/src/network/gossip/validation/attestation.ts

```diff
@@ -120,20 +157,28 @@ export async function validateGossipAttestation(
 
 export async function isAttestingToInValidBlock(db: IBeaconDb, attestation: Attestation): Promise<boolean> {
   const blockRoot = attestation.data.beaconBlockRoot.valueOf() as Uint8Array;
-  //TODO: check if source and target blocks are not in bad block repository
+  // TODO: check if source and target blocks are not in bad block repository
   return await db.badBlock.has(blockRoot);
 }
 
-/**
- * is ready to be included in block
- */
-export function hasValidAttestationSlot(config: IBeaconConfig, genesisTime: number, attestation: Attestation): boolean {
-  const currentSlot = getCurrentSlot(config, genesisTime);
+export function isUnaggregatedAttestation(attestation: Attestation): boolean {
+  return Array.from(attestation.aggregationBits).filter((bit) => !!bit).length === 1;
+}
+
+export function isCommitteeIndexWithinRange(epochCtx: EpochContext, attestationData: AttestationData): boolean {
+  return attestationData.index < epochCtx.getCommitteeCountAtSlot(attestationData.target.epoch);
+}
+
+export function doAggregationBitsMatchCommitteeSize(
+  attestationPreStateContext: ITreeStateContext,
+  attestation: Attestation
+): boolean {
   return (
-    attestation.data.slot + ATTESTATION_PROPAGATION_SLOT_RANGE >= currentSlot && currentSlot >= attestation.data.slot
+    attestation.aggregationBits.length ===
+    attestationPreStateContext.epochCtx.getBeaconCommittee(attestation.data.slot, attestation.data.index).length
   );
 }
 
-export function isUnaggregatedAttestation(attestation: Attestation): boolean {
-  return Array.from(attestation.aggregationBits).filter((bit) => !!bit).length === 1;
+export function doesEpochSlotMatchTarget(config: IBeaconConfig, attestationData: AttestationData): boolean {
+  return attestationData.target.epoch === computeEpochAtSlot(config, attestationData.slot);
```

Comment:
@wemeetagain Should we use the SSZ equality helper here in case we switch types?